src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -1240,89 +266,113 @@ def main():
+                    ),
+                    sort=eodag_sort,
+                    provider=eodag_provider,
+                    pattern=eodag_pattern,
+                    footprints=options["footprints"] if options["footprints"] else None,
+                    quiet=True,
+                )
+            )
+        except CalledModuleError:
+            gs.fatal(
+                _(
+                    "Could not connect to {}.\nPlease check your credentials or try again later.".format(
+                        options["datasource"]
+                    )
+                )
+            )
 
     if flags["l"]:
-        downloader.list()
-        return
-
-    downloader.download(
-        options["output"],
-        options["sleep"],
-        int(options["retry"]),
-        options["datasource"],
-    )
-
+        # TODO: Implement listing
+        pass
+    else:
+        if len(scenes) == 0:
+            return
+        create_dir(options["output"])
+        gs.run_command(
+            "i.eodag",
+            producttype=eodag_producttype,
+            output=options["output"],
+            map=options["map"] if options["map"] else None,
+            start=start_date,
+            end=end_date,
+            clouds=options["clouds"] if options["clouds"] else None,
+            limit=options["limit"],
+            order=options["order"],
+            sort=eodag_sort,
+            provider=options["datasource"],
+            pattern=eodag_pattern,
+            quiet=True,
+        )
     return 0
 
 
 if __name__ == "__main__":
     options, flags = gs.parser()
 
-    if options["datasource"] == "GCS":
-        # Lazy import nonstandard modules
-        try:
-            import pandas
-        except ImportError as e:
-            gs.fatal(_("Module requires pandas library: {}").format(e))
-
-        try:
-            import requests
-        except ImportError as e:
-            gs.fatal(_("Module requires requests library: {}").format(e))
+    try:
+        import eodag
+        from eodag import EODataAccessGateway
+    except:
+        gs.fatal(_("Cannot import eodag. Please intall the library first."))
 
-        try:
-            from tqdm import tqdm
-        except ImportError as e:
-            gs.fatal(_("Module requires tqdm library: {}").format(e))
-    elif options["datasource"] == "USGS_EE":
-        try:
-            import landsatxplore.api
-            from landsatxplore.errors import EarthExplorerError
-        except ImportError as e:
-            gs.fatal(_("Module requires landsatxplore library: {}").format(e))
+    try:
+        gs.find_program("i.eodag", "--help")
+    except ImportError:
+        gs.fatal(_("Addon i.eodag not found. Please intall it with g.extension."))
+
+    # Check if eodag config file is created
+    file_path = os.path.join(os.path.expanduser("~"), ".config/eodag/eodag.yml")
+    if not os.path.isfile(file_path):
+        dag = EODataAccessGateway()
+        gs.info(
```

Comment:
@HamedElgizery are you sure this is needed? is the config file already exists this is superfluous

Reply:
I think it is important because Copernicus Data Space Ecosystem, which has almost all the products, requires credentials for searching. So it can be an indication for the user to fill in the credentials.
However, I think having the settings file option back is the best workaround, but I need to implement it in i.eodag first...

Reply:
> @HamedElgizery are you sure this is needed? is the config file already exists this is superfluous

I found a way to have the settings option back, without needing to add anything to i.eodag.
I have put back the settings option, and removed the config file check in the last couple commits.